ministryofjustice/laa-review-criminal-legal-aid
# [CRIMAPP-374] Application history changes for pse applications

## Description of change
Amends the application event history to display the correct content for pse application events
Also prevents parent applications of pse applications from being generated a superseded event

## Link to relevant ticket
[CRIMAPP-374](https://dsdmoj.atlassian.net/browse/CRIMAPP-374)

## Notes for reviewer
The application history screenshots below show an event for marking the pse as ready for assessment which can be ignored. There is an upcoming ticket to change the review action buttons for pse applications which will remove the mark as ready for assessment button and so it won't show on the application history. 

## Screenshots of changes (if applicable)

### Before changes:

### After changes:
<img width="816" alt="Screenshot 2024-01-25 at 15 31 41" src="https://github.com/ministryofjustice/laa-review-criminal-legal-aid/assets/51047911/df5dc912-7946-4422-a3bd-6a3ad019bed7">

## How to manually test the feature

**Note: Make sure you have built datastore from main and crime apply from this [branch](https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/pull/616)** 

- Submit a standard application in apply and review the application in review (as can only submit pse for applications that are complete)
- Then go back to apply and find the application in the submitted tab. Click on the application and click upload post submission evidence 
- Submit your pse application and then go to review and find the application and check there is a `Post submission evidence submitted` event 
- Also check the following pse events: 
	- Assign the pse to yourself and check the assign event
	- Remove the pse from your list and check the removed event 
	- Mark the pse as complete and check the complete event 

Also worth checking the regular return/resubmission events display as expected

[CRIMAPP-374]: https://dsdmoj.atlassian.net/browse/CRIMAPP-374?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
